### PR TITLE
Pipelines 1.6.3 Release Notes

### PR DESCRIPTION
**Pull request by sounix000**
- **Aligned team**: Dev Tools
- **OCP version for cherry-picking**: `enterprise-4.9`, `enterprise-4.10`, `enterprise-4.11`
- **JIRA issues**: [RHDEVDOCS-4070 RN and Errata for Pipelines 1.6.3 and 1.7.1](https://issues.redhat.com/browse/RHDEVDOCS-4070)
- **Preview pages**: No preview generated due to Netlify issues.
- **SME review**: @concaf       
- **Peer-review**: @Srivaralakshmi

### CODE REVIEW

**Review by concaf — APPROVED** [review]

File: modules/op-release-notes-1-6.adoc [hunk 1]
```diff
@@ -306,4 +306,32 @@ Error from server (InternalError): Internal error occurred: failed calling webho
...
+=== Fixed issues
+
+* With this update, project administrators can access pipeline metrics in the Developer console.
+// https://issues.redhat.com/browse/SRVKP-2129
+
+* With this update, the reliability and consistency of installing various components of {pipelines-title} are improved.
+// https://issues.redhat.com/browse/SRVKP-2249
+
+* With this update, the speed of lifecycle management of Tekton resources are improved.
+// https://issues.redhat.com/browse/SRVKP-2043
```
Comment (Srivaralakshmi) [inline modules/op-release-notes-1-6.adoc:337]: None of this mentions what was the case before the update. Please rewrite following the guideline prescribed here: https://redhat-documentation.github.io/supplementary-style-guide/#release-notes

**Review by Srivaralakshmi — CHANGES_REQUESTED** [review]
@sounix000 Left a suggestion. PTAL. Rest looks good to me. Thanks!

File: modules/op-release-notes-1-6.adoc [hunk 2]
```diff
@@ -306,4 +306,32 @@ Error from server (InternalError): Internal error occurred: failed calling webho
...
+[NOTE]
+====
+When you upgrade to future releases, the {pipelines-title} Operator will automatically delete all obsolete pod security policies.
+====
+// https://issues.redhat.com/browse/SRVKP-2259
+
+[id="fixed-issues-1-6-3_{context}"]
+=== Fixed issues
+
+* Before this update, only cluster administrators could access pipeline metrics in the {product-title} console. With this update, users with other cluster roles can access the pipeline metrics.
```
Comment (Srivaralakshmi) [inline modules/op-release-notes-1-6.adoc:330]:
```suggestion
* Before this update, only cluster administrators could access pipeline metrics in the {product-title} console. With this update, users with other cluster roles also can access the pipeline metrics.
```

File: modules/op-release-notes-1-6.adoc [hunk 3]
```diff
@@ -306,4 +306,32 @@ Error from server (InternalError): Internal error occurred: failed calling webho
...
+
+[id="release-notes-1-6-3_{context}"]
+== Release notes for {pipelines-title} General Availability 1.6.3
+
+[id="deprecated-features-1-6-3_{context}"]
+=== Deprecated features
+
+* With this update, the {pipelines-title} Operator stops installing pod security policies from components, as the policies are deprecated. As a result, the following upgrade paths are affected:
+** When you upgrade from {pipelines-shortname} 1.6.1 or 1.6.2 to {pipelines-shortname} 1.6.3, the existing pod security policies from components such as Pipelines and Triggers are deleted.
+** When you upgrade from {pipelines-shortname} 1.5.x to 1.6.3, the existing pod security policies are retained. However, cluster administrators can delete them manually.
```
Comment (Srivaralakshmi) [inline modules/op-release-notes-1-6.adoc:319]: Ensure you follow this guideline mentioned here for Deprecated functionality: https://redhat-documentation.github.io/supplementary-style-guide/#release-notes

File: modules/op-release-notes-1-6.adoc [hunk 4]
```diff
@@ -306,4 +306,32 @@ Error from server (InternalError): Internal error occurred: failed calling webho
...
+====
+// https://issues.redhat.com/browse/SRVKP-2259
+
+[id="fixed-issues-1-6-3_{context}"]
+=== Fixed issues
+
+* Before this update, only cluster administrators could access pipeline metrics in the {product-title} console. With this update, users with other cluster roles can access the pipeline metrics.
+// https://issues.redhat.com/browse/SRVKP-2129
+
+* Before this update, multiple implementation issues in RBAC caused rare problems during installation and upgrade of components shipped with {pipelines-shortname} Operator. With this update, the reliability and consistency of installing various components of {pipelines-title} are improved.
```
Comment (Srivaralakshmi) [inline modules/op-release-notes-1-6.adoc:333]:
```suggestion
* Before this update, multiple implementation issues in role-based access control (RBAC) caused rare problems during the installation and upgrade of components shipped with {pipelines-shortname} Operator. With this update, the reliability and consistency of installing various components of {pipelines-title} are improved.
```

File: modules/op-release-notes-1-6.adoc [hunk 5]
```diff
@@ -306,4 +306,32 @@ Error from server (InternalError): Internal error occurred: failed calling webho
...
+// https://issues.redhat.com/browse/SRVKP-2249
+
+* Before this update, setting the `clusterTasks` and `pipelineTemplates` fields to `false` in the `TektonConfig` CR resulted in slow removal of cluster tasks and pipeline templates. With this update, the speed of lifecycle management of Tekton resources such as cluster tasks and pipeline templates have improved.
```
Comment (Srivaralakshmi) [inline modules/op-release-notes-1-6.adoc:336]:
```suggestion
* Before this update, setting the `clusterTasks` and `pipelineTemplates` fields to `false` in the `TektonConfig` CR resulted in the slow removal of cluster tasks and pipeline templates. With this update, the speed of lifecycle management of Tekton resources such as cluster tasks and pipeline templates has improved.
```

**Review by Srivaralakshmi — CHANGES_REQUESTED** [review]
@sounix000 Left some suggestions. PTAL.

I have used the OpenShift peer review checklist.

File: modules/op-release-notes-1-6.adoc (hunk 3 above)
Comment (sounix000) [reply modules/op-release-notes-1-6.adoc]: This is not exactly a deprecated feature. This is actually a fixed issue that was overdue for fix after something was deprecated in an earlier release. Accordingly, I have restructured the sentence and included it under Fixed issues.

**Comment by sounix000** [comment]
I have followed the [OpenShift Peer Review Checklist > (for writers)](https://docs.google.com/document/d/1WN7k72PKxcPA__erp5TK-CefmS_sVwE3r40PvpmNk3E/edit#heading=h.bp4j7064auhh)

File: modules/op-release-notes-1-6.adoc (hunk 3 above)
Comment (Srivaralakshmi) [reply modules/op-release-notes-1-6.adoc]: @sounix000 Thanks for the explanation. this helps :)

**Review by Srivaralakshmi — CHANGES_REQUESTED** [review]
@sounix000 Thanks for fixing the comments. some minor nitpicks. Otherwise LGTM.

**Review by Srivaralakshmi — APPROVED** [review]
@sounix000 Approved! Thanks for the changes.

File: modules/op-release-notes-1-6.adoc [hunk 6]
```diff
@@ -306,4 +306,29 @@ Error from server (InternalError): Internal error occurred: failed calling webho
...
+// https://github.com/tektoncd/pipeline/releases/tag/v0.28.3
+
+[id="release-notes-1-6-3_{context}"]
+== Release notes for {pipelines-title} General Availability 1.6.3
+
+[id="fixed-issues-1-6-3_{context}"]
+=== Fixed issues
+
+* Before this update, the {pipelines-title} Operator installed pod security policies from components such as Pipelines and Triggers. However, the pod security policies shipped as part of the components were deprecated in an earlier release. With this update, the Operator stops installing pod security policies from components. As a result, the following upgrade paths are affected:
+** When you upgrade from {pipelines-shortname} 1.6.1 or 1.6.2 to {pipelines-shortname} 1.6.3, the existing pod security policies from components such as Pipelines and Triggers are deleted.
```
[the remark this answers is not in the document]
Comment (rolfedh) [reply modules/op-release-notes-1-6.adoc:318]: - Does upgrading delete the security policies from any component (first suggestion), or only the security policies from Pipelines and Triggers (second suggestion)? 
```suggestion
** Upgrading from {pipelines-shortname} 1.6.1 or 1.6.2 to {pipelines-shortname} 1.6.3 deletes the pod security policies, including those from the Pipelines and Triggers components.
```
```suggestion
** Upgrading from {pipelines-shortname} 1.6.1 or 1.6.2 to {pipelines-shortname} 1.6.3 deletes the pod security policies from the Pipelines and Triggers components.
```
Sometimes, if it doesn't add clarity, you can drop `existing` as redundant.

File: modules/op-release-notes-1-6.adoc [hunk 7]
```diff
@@ -306,4 +306,29 @@ Error from server (InternalError): Internal error occurred: failed calling webho
...
+** When you upgrade from {pipelines-shortname} 1.5.x to 1.6.3, the existing pod security policies are retained. However, cluster administrators can delete them manually.
```
Comment (rolfedh) [inline modules/op-release-notes-1-6.adoc:319]:
```suggestion
** Upgrading from {pipelines-shortname} 1.5.x to 1.6.3 retains the pod security policies. As a cluster administrator, delete these pod security policies by using the `[TBD]` command.
```
- You can remove "However." It creates a false opposition with the first sentence.
- Do we need to clarify whether the policies are still in-effect or tell the administrator to delete them?

File: modules/op-release-notes-1-6.adoc [hunk 8]
```diff
@@ -306,4 +306,29 @@ Error from server (InternalError): Internal error occurred: failed calling webho
...
+[NOTE]
+====
+When you upgrade to future releases, the {pipelines-title} Operator will automatically delete all obsolete pod security policies.
+====
+// https://issues.redhat.com/browse/SRVKP-2259
+
+* Before this update, only cluster administrators could access pipeline metrics in the {product-title} console. With this update, users with other cluster roles also can access the pipeline metrics.
+// https://issues.redhat.com/browse/SRVKP-2129
+
+* Before this update, multiple implementation issues in role-based access control (RBAC) caused rare problems during the installation and upgrade of components shipped with {pipelines-shortname} Operator. With this update, the reliability and consistency of installing various components of {pipelines-title} are improved.
```
Comment (rolfedh) [inline modules/op-release-notes-1-6.adoc:330]:
```suggestion
* Before this update, role-based access control (RBAC) issues with the {pipelines-shortname} Operator caused problems upgrading or installing components. This update improves the reliability and consistency of installing various {pipelines-title} components.
```

File: modules/op-release-notes-1-6.adoc [hunk 9]
```diff
@@ -306,4 +306,29 @@ Error from server (InternalError): Internal error occurred: failed calling webho
...
+// https://issues.redhat.com/browse/SRVKP-2249
+
+* Before this update, setting the `clusterTasks` and `pipelineTemplates` fields to `false` in the `TektonConfig` CR resulted in the slow removal of cluster tasks and pipeline templates. With this update, the speed of lifecycle management of Tekton resources such as cluster tasks and pipeline templates has improved.
```
Comment (rolfedh) [inline modules/op-release-notes-1-6.adoc:333]:
```suggestion
* Before this update, setting the `clusterTasks` and `pipelineTemplates` fields to `false` in the `TektonConfig` CR slowed the removal of cluster tasks and pipeline templates. This update improves the speed of lifecycle management of Tekton resources such as cluster tasks and pipeline templates.
```

File: modules/op-release-notes-1-6.adoc [hunk 10]
```diff
@@ -77,7 +77,7 @@ The `keep` and `keep-since` annotations are mutually exclusive. For any resource
 
 * Administrators can disable the creation of the `pipeline` service account for the entire cluster, and prevent privilege escalation by misusing the associated SCC, which is very similar to `anyuid`.
 
-* You can now configure feature flags and components by using the `TektonConfig` custom resource, and the custom resources for individual components, such as `TektonPipeline` and `TektonTriggers`. This level of granularity is useful for customization and testing of alpha features such as the Tekton OCI bundle for individual components.
+* You can now configure feature flags and components by using the `TektonConfig` custom resource (CR), and the custom resources (CRs) for individual components, such as `TektonPipeline` and `TektonTriggers`. This level of granularity is useful for customization and testing of alpha features such as the Tekton OCI bundle for individual components.
```
[the remark this answers is not in the document]
Comment (rolfedh) [reply modules/op-release-notes-1-6.adoc:80]: Maybe "...helps you..."

**Comment by Preeticp** [comment]
followed the [OpenShift Peer Review Checklist > (for mergers)](https://docs.google.com/document/d/1WN7k72PKxcPA__erp5TK-CefmS_sVwE3r40PvpmNk3E/edit#heading=h.qeeeis1fmrjg)

**Pull request merged**